ucs/database.cpp

```diff
@@ -280,16 +304,136 @@ bool Database::LoadChatChannels()
 	}
 
 	for (auto row = results.begin(); row != results.end(); ++row) {
-		std::string channelName     = row[0];
-		std::string channelOwner    = row[1];
+		std::string channelName = row[0];
+		std::string channelOwner = row[1];
 		std::string channelPassword = row[2];
 
-		ChannelList->CreateChannel(channelName, channelOwner, channelPassword, true, atoi(row[3]));
-	}
 
+		if (IsOnChannelBlockList(channelName)) {
+			if (!(channelOwner == "*Block*")) { // Dont show that channels were blocked when the block entries themselves are detected.
+				LogDebug("Blocked channel [{}] load from database due to being blocked by server operator.", channelName);
+			}
+		}
+		else {
+			ChannelList->CreateChannel(channelName, channelOwner, channelPassword, true, atoi(row[3]), false);
+		}
+	}
 	return true;
 }
 
+bool Database::IsOnChannelBlockList(std::string channelName) {
+	// Ignore the input if it is empty
+	if (channelName == "") {
+		return false;
+	}
+	// Convert channelName to lowercase for case-insensitive comparison
```

Comment:
We have 

```cpp
std::string Strings::ToLower(std::string s)
```

For converting strings to lower

Reply:
Updated